sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -497,6 +497,27 @@ public void queryPlanDiagnostics() throws JsonProcessingException {
         }
     }
 
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void readManyDiagnostics() {
+        String pkValue = UUID.randomUUID().toString();
+        PartitionKey partitionKey = new PartitionKey(pkValue);
+        CosmosContainer cosmosContainer = directClient.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+        List<CosmosItemIdentity> itemIdList = new ArrayList<>();
+        for(int i = 0; i< 100; i++) {
+            InternalObjectNode internalObjectNode = getInternalObjectNode(pkValue);
+            CosmosItemResponse<InternalObjectNode> createResponse = cosmosContainer.createItem(internalObjectNode);
+            if(i%20 == 0) {
+                itemIdList.add(new CosmosItemIdentity(partitionKey, internalObjectNode.getId()));
+            }
+        }
+
+        FeedResponse<InternalObjectNode> response = cosmosContainer.readMany(itemIdList, InternalObjectNode.class);
+        FeedResponseDiagnostics diagnostics = response.getCosmosDiagnostics().getFeedResponseDiagnostics();
+
+        assertThat(diagnostics.getClientSideRequestStatisticsList().size()).isEqualTo(itemIdList.size());
+        assertThat(diagnostics.getQueryMetricsMap().values().iterator().next().getRetrievedDocumentCount()).isEqualTo(itemIdList.size());
```

Comment:
Is there an idiomatic Java way to get the first element of a `Collection` like .NET's `First()` extension method?

Reply:
Not really - with streams there are - but I guess in this case what you have is the most common pattern and the easiest to read/understand.